### ISSUE TITLE
Rewrite examples in `printf` wherever possible?

### ISSUE THREAD

**Issue by gsklee** [anordal/shellharden#26]
@anordal In [echo → printf](https://github.com/anordal/shellharden/blob/master/how_to_do_things_safely_in_bash.md#echo--printf), the guide is trying to make a case that `printf` should be preferred over `echo`; however in a lot of other places within the guide (eg. [Corollary: Use while loops to iterate strings and command output](https://github.com/anordal/shellharden/blob/master/how_to_do_things_safely_in_bash.md#corollary-use-while-loops-to-iterate-strings-and-command-output)), `echo` is being used extensively. Do we want to consolidate the guide and rewrite examples in `printf` wherever possible?

**Comment by gsklee** [comment]
@anordal Any explanations?

**Comment by anordal** [comment]
It should always be possible to replace `echo` with `printf`. I only rewrote the guide's examples in `printf` where necessary according to itself – I admit it was inconsistent. I also tried to de-emphasize the hysteria and focus on the *why* part (5f425ee47).

Are you wondering why I didn't change the rest?

In the cases where it doesn't matter, I have two concerns:
* The verbosest alternative is usually hard to defend. One of my heroes, Jussi Pakkanen argues that it is [human nature to always pick the simple solution even when it is the wrong one](http://voices.canonical.com/jussi.pakkanen/2014/07/22/the-two-ways-of-doing-something/).
* For example purposes, verbosity is distraction too.
